src/syntax/Scanner.js

```diff
@@ -1172,11 +1172,33 @@ function isAtEnd() {
 
 function next() {
   index++;
-  updateCurrentCharCode();
+  if (currentCodePoint >= 0x10000) {
+    // If code point is higher than Plane 0, move past trail surrogate
+    index++;
+  }
+  updateCurrentCodePoint();
+}
+
+function updateCurrentCodePoint() {
+  currentCodePoint = input.charCodeAt(index);
+  // Return if code point is in Plane 0
+  if (currentCodePoint < 0xD800 || !isLeadSurrogate(currentCodePoint)) return;
```

Comment:
fast case: U+D7FF is a pretty big range of codes, and is the last code point before a trail of lead surrogates --- slightly slower are at the higher end of the BMP, which are detected with the function call